### PR TITLE
Issue/az 53 respond to spam events

### PR DESCRIPTION
**Pull request by eugene-o-z**

### CODE REVIEW

File: Tests/Services/AzineMailgunMailerServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    /**
+     * Check if the current setup is a full application.
+     * If not, mark the test as skipped else continue.
+     */
+    private function checkApplication()
+    {
+        try {
+            static::$kernel = static::createKernel(array());
+        } catch (\RuntimeException $ex) {
+            $this->markTestSkipped("There does not seem to be a full application available (e.g. running tests on travis.org). So this test is skipped.");
```
Comment (alister) [inline Tests/Services/AzineMailgunMailerServiceTest.php]: We need an AppKernel for running tests. At the moment, I not sure that any tests with this `checkApplication()` method, that actually fetch services are even capable of being run.

Comment (azine) [reply Tests/Services/AzineMailgunMailerServiceTest.php]: Yes. I know. That's the whole point of the `checkApplication()` test. This way you can run the tests on your computer, but they won't fail on travis-ci.

**Comment by azine** [comment]
About PHP-CS-Fixer : I agree. I'll try to get this integrated into the build-process. I think you can configure it as "build checks", right?! I'll look into it.

File: Services/AzineMailgunMailerService.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                ->setFrom($this->mailerUser)
+                ->setSubject($this->translator->trans('notification.spam_complaint_received'))
+                ->setBody(
+                    $this->twig->render('@AzineMailgunWebhooks/Email/notification.html.twig', array('eventId' => $eventId, 'ticketId' => $this->ticketId)),
+                    'text/html'
+                );
+
+            $lastSpamReport = $this->entityManager->getRepository(EmailTrafficStatistics::class)
+                ->findOneBy(['action' => EmailTrafficStatistics::SPAM_ALERT_SENT],
+                    ['created' => 'DESC']);
```
Comment (azine) [inline Services/AzineMailgunMailerService.php]: sorting a "findOneBy" does not do anything

File: Services/AzineMailgunMailerService.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            $message = $this->mailer->createMessage();
+            $message->setTo($this->adminUserEmail)
+                ->setFrom($this->mailerUser)
+                ->setSubject($this->translator->trans('notification.spam_complaint_received'))
+                ->setBody(
+                    $this->twig->render('@AzineMailgunWebhooks/Email/notification.html.twig', array('eventId' => $eventId, 'ticketId' => $this->ticketId)),
+                    'text/html'
+                );
+
+            $lastSpamReport = $this->entityManager->getRepository(EmailTrafficStatistics::class)
```
Comment (azine) [inline Services/AzineMailgunMailerService.php]: move this logic to the repository => 
`$lastSpamReport = $this->entityManager->getRepository(EmailTrafficStatistics::class)->findEmailTrafficStatisticsSentWithin($minutes)`

File: Services/AzineMailgunMailerService.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                $timeDiff = $time->diff($lastSpamReport->getCreated());
+
+                if($timeDiff->i > $this->sendNotificationsInterval) {
+
+                    $messagesSent = $this->mailer->send($message);
+                }
+
+            }else{
+
+                $messagesSent = $this->mailer->send($message);
```
Comment (azine) [inline Services/AzineMailgunMailerService.php]: throw exception if sending fails

File: Resources/translations/messages.en.xliff [hunk 5]
```diff
@@ -10,6 +10,18 @@
         <source>dummy</source>
         <target>dummy</target>
       </trans-unit>
+      <trans-unit id="0b5e2fc2fbb589463a61d536b0d97fa25028bbfd" resname="notification.spam_complaint_received">
+        <source>notification.spam_complaint_received</source>
```
Comment (azine) [inline Resources/translations/messages.en.xliff]: please add the `state="new"` for all new translations

File: Tests/Services/AzineMailgunMailerServiceTest.php (hunk 1 above)
Comment (alister) [reply Tests/Services/AzineMailgunMailerServiceTest.php]: I'll see about adding a minimal AppKernel as part of the tests to provide a suitable environment & container.

**Comment by azine** [comment]
Merged after rebase https://github.com/azine/AzineMailgunWebhooksBundle/pull/26

File: Resources/translations/messages.en.xliff (hunk 5 above)
Comment (alister) [reply Resources/translations/messages.en.xliff]: I've just updated my (currently Symfony 3.3)  project, and now every page is throwing the error:

An exception has been thrown during the rendering of a template 
("Invalid resource provided: "1.2"; Errors: [ERROR 1867] 
Element '{urn:oasis:names:tc:xliff:document:1.2}
source', attribute 'state': The attribute 'state' is not allowed. (in /home/.../web/ - line 14, column 0)

The [Xliff schema](http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#state) does  mention 'state', but it's not on `source`, its on the next element - `target`.

Comment (azine) [reply Resources/translations/messages.en.xliff]: ups sorry. I'll fix that later today.

**Pull request closed without merging**